ipq807x: disable Malibu PHY in favor of upstream version

Ansuel opened the pull request:
Move qca807x from ipq40xx to generic to use also on ipq807x and also add LEDs support for it.

robimarko commented on package/kernel/qca-ssdk/Makefile:
```diff
@@ -48,7 +48,7 @@ MAKE_FLAGS+= \
 
 ifeq ($(CONFIG_TARGET_SUBTARGET), "ipq807x")
-    MAKE_FLAGS+= CHIP_TYPE=HPPE
+    MAKE_FLAGS+= CHIP_TYPE=HPPE IN_MALIBU_PHY=FALSE
```
I would move this to the rest of make flags as otherwise we will have to disable it in ipq60xx, then ipq50xx etc

Ansuel committed:
qualcommax: set correct PHY mode for port 0-4

Port 0-4 have the mode set to SGMII instead of PSGMII. Now that we use
the upstream qca807x driver, this conflicts with the qca SSDK driver
that expects the mode to be PSGMII as for not integrated driver, it does
refer to the real PHY mode.

Update the entry for port 0-4 to PSGMII to solve warning from qca SSDK
in ipq8074-ess.dtsi.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qca-ssdk: add patch to support detection of PSGMII mode for PHY

If a PHY doesn't use the integrated driver, SSDK use poll the phydev to
get the real PHY mode. qca807x use PSGMII as PHY mode and this specific
mode is not detected in qca SSDK while used in the entire driver.

Add support for it in the hsl_port_phydev_interface_mode_status_get
function used to translate PHY mode to the internal SSDK value.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qca-ssdk: disable Malibu PHY in favor of upstream version

Disable Malibu PHY driver in Qca SSDK in favor of the upstream version.
The same workaround are applied and the version upstream is just a drop
in replacement and is well tested from the ipq40xx target.

Also using the upstream version permits further support for LEDs.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommax: enable qca807x PHY driver

IPQ807x have integrated qca8074 PHY supported by the upstream qca807x driver.

Enable it to use it instead of the downstream qca SSDK variant.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommax: apply special PHY LEDs configuration for Xiaomi AX9000

Xiaomi AX9000 apply a special PHY LEDs configuration where the unique
green LED for each qca807x PHY port is turned on also on 1000Mbps link.

Apply this special configuration to reflect original implementation.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommax: apply special PHY LEDs configuration for Xiaomi AX3600

Xiaomi AX3600 apply a special PHY LEDs configuration where the unique
green LED for each qca807x PHY port is turned on also on 1000Mbps link.

Apply this special configuration to reflect original implementation.

Also enable CONFIG_PHYLIB_LEDS to actually expose the PHY LEDs if
defined in DT.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qca-ssdk: drop deprecated Xiaomi LEDs quirk patch

Drop deprecated Xiaomi LEDs quirk patches as they are not needed anymore
as LEDs are now supported by the upstream qca807x driver.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

aiamadeus commented:
Just curious, since kernel has the qca8081 driver, can we use it?

Ansuel commented:
@aiamadeus yep robi tested that on ax9000

robimarko commented:
@Ansuel We need to add the PSGMII PHY inside of QCA807x to the DTS so that quirks for it get applied.
It shows up as 5th MDIO device in the PHY with the ID of `0x06820805`

So for cases where QCA807x start addr is 0, its just on address 5 so this is enough in the MDIO node:
```
qca8075_psgmii: ethernet-phy@5 {
		compatible = "ethernet-phy-ieee802.3-c22";
		reg = <5>;
	};
```

Ansuel commented:
Guess for 301w is different

robimarko commented:
Yes, Qnap has the PSGMII analog PHY at `0x15`

Ansuel committed:
qualcommax: convert qca807x PHY to PHY package implementation

Convert every qca807x PHY definition in DT to new PHY package
implementation to correctly support applying fixup for the correct PHY
mode.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

robimarko commented:
@Ansuel Can you mark this as draft for now, since the upstream discussion is not yet complete.

aiamadeus commented:
I'm confused, commit 7f0f838 broke the wan port on aw1000.
If uboot initializes the network, the wan port can work as before.
```
nss-dp 3a007000.dp6-syn wan: PHY Link up speed: 2500
```
If uboot does not initialize the network (boot without interrupting),
then the wan port can't work. At this time, the phy id of qca8081
can be seen on the mdio bus.
```
mdio 90000.mdio-1
 DEV      PHY-ID  LINK
0x00  0x004dd0b1  up
0x01  0x004dd0b1  down
0x02  0x004dd0b1  down
0x03  0x004dd0b1  down
0x04  0x004dd0b1  down
0x05  0x04820a05  down
0x1c  0x004dd101  down
0x1d  0x8aa21ac8  down
```

Ansuel commented:
@aiamadeus can you open an issue so we can track that there?
